### ISSUE TITLE
Improve documentation for developing an extension against the JupyterLab source code in dev mode

### ISSUE THREAD

**Issue by hbcarlos** [jupyterlab/jupyterlab#9896]
### Problem
When developing a prebuilt extension against the JupyterLab source repo using the options `--dev-mode` `--extensions-in-dev-mode`, the Typescript compiler complains about types.

This is due to the extension is getting types from its `node_modules` folder while imported packages are getting types from JupyterLab's `node_modules` folder. The Typescript compiler detects two files with the same type and throws an error.

### Proposed Solution
Add this case to the documentation in [developing a prebuilt extension section](https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html?#developing-a-prebuilt-extension).

I solved this problem by adding to my `tsconfig.json` in `compilerOptions`, the path of the different packages that I'm using in my extension.
```json
"compilerOptions": {
    ...,
    "baseUrl": ".",
    "paths": {
        "@jupyterlab/application": ["./relative/path/to/jupyterlab/packages/application"],
        ...,
        "*": ["node_modules/*"]
    }
}
```

### Additional context
I was using a local [JupyterLab](https://github.com/QuantStack/jupyterlab/tree/yjupyter) with packages that differ from the published ones.

**Comment by jtpio** [comment]
Thanks @hbcarlos.

We can probably iterate on this section of the documentation to add more details and hints: 

https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html?#developing-a-prebuilt-extension

Also for reference there is a bit of explanations in the help of `--extensions-in-dev-mode`: https://github.com/jupyterlab/jupyterlab/blob/3385b963ec321d4579df47e3cc8379d12f8e9cf9/jupyterlab/labapp.py#L528-L533

**Comment by hbcarlos** [comment]
Solved in #10190
